### PR TITLE
basic venn

### PR DESCRIPTION
**Pull request by craigmichaelmartin**
@CINBCUniversal/suprematism

### CODE REVIEW

File: src/venn-diagram.component.ts [hunk 1]
```diff
@@ -1,10 +1,108 @@
...
+  encapsulation: ViewEncapsulation.None,
   styles: [require('./venn-diagram.component.scss')]
 })
-export class VennDiagramComponent {
-  title = 'app works!';
+export class VennDiagramComponent implements AfterViewInit, OnDestroy, OnChanges {
+
+  @Input() vennSets: Array<VennSet>;
+  tooltip: any;
+  div: any;
```
Comment (xerona) [inline src/venn-diagram.component.ts]: Is there something more specific than `any`.

**Comment by caffeinelogic** [comment]
We still need the API support before this can be merged? Or is this making some assumptions and we'll adjust when we get it like we have in the past?

File: src/venn-diagram.component.ts (hunk 1 above)
Comment (caffeinelogic) [inline src/venn-diagram.component.ts]: Agree, can we be more specific?

File: package.json [hunk 2]
```diff
@@ -36,7 +37,9 @@
     "@angular/platform-browser-dynamic": "^4.0.0",
     "@angular/router": "^4.0.0",
     "core-js": "^2.4.1",
+    "d3": "^4.7.4",
```
Comment (caffeinelogic) [inline package.json]: This is throwing an error when I bring it into the Audri v4 tes repo. I think this needs to be a dependency, not a dev one.

File: package.json [hunk 3]
```diff
@@ -54,6 +57,7 @@
     "@types/node": "~6.0.60",
     "codelyzer": "~2.0.0",
     "core-js": "^2.4.1",
+    "d3": "^4.7.4",
```
Comment (caffeinelogic) [inline package.json]: This is throwing an error when I bring it into the Audri v4 test repo. I think this needs to be a dependency, not a dev one.

File: package.json [hunk 4]
```diff
@@ -67,6 +71,7 @@
     "ts-node": "~2.0.0",
     "tslint": "~4.5.0",
     "typescript": "~2.2.0",
+    "venn.js": "^0.2.14",
```
Comment (caffeinelogic) [inline package.json]: This is throwing an error when I bring it into the Audri v4 test repo. I think this needs to be a dependency, not a dev one.

File: package.json (hunk 3 above)
Comment (craigmichaelmartin) [inline package.json]: It is listed as a dev dep for the local example demo. I list it as a peer dep above also. My thought was that if we have more components that use d3, we don't want each component pulling it its own d3 dep, but sharing the one in the app. Thoughts?

**Comment by craigmichaelmartin** [comment]
We don't need API support before this can be merged; just before it can be pulled into the audri app. I am holding off incorporating it into the app until API support is there.

File: src/venn-diagram.component.ts (hunk 1 above)
Comment (craigmichaelmartin) [inline src/venn-diagram.component.ts]: Yep, thanks.

**Comment by craigmichaelmartin** [comment]
@caffeinelogic @xerona ready

**Comment by xerona** [comment]
lgtm

**Pull request merged**